cmd/tag-show.go

```diff
@@ -0,0 +1,215 @@
+func parseTagMessage(tagsStr string, urlStr string) tagSetMessage {
+	var t tagSetMessage
+	var tagStr string
+	tagStr = strings.Replace(tagsStr, "\\u0026", "&", -1)
+	kvPairStr := strings.Split(tagStr, "&")
+
+	for _, kvPair := range kvPairStr {
+		kvPairSplit := splitStr(kvPair, "=", 2)
+		t.Tags = append(t.Tags, tagMessage{Key: kvPairSplit[0], Value: kvPairSplit[1]})
+	}
+
+	return t
+}
+
+func showTagObjectName(urlStr string) {
+	clnt, pErr := newClient(urlStr)
+
+	if pErr != nil {
+		console.Errorln(pErr.ToGoError().Error)
+		fatalIf(probe.NewError(errors.New("Unable to show tags")), "Unable to obtain client from provided url "+urlStr)
+	}
+	s3c, ok := clnt.(*s3Client)
+	if !ok {
+		fatalIf(probe.NewError(errors.New("Unable to show tags")), "Unable to show tags for object in URL "+urlStr)
+	}
+	bucket, object := s3c.url2BucketAndObject()
+	objectName := fmt.Sprintf("%-10s: %s", "Object", bucket+slashSeperator+object)
+	console.Println(console.Colorize(tagMainHeader, objectName))
+}
+
+func getTagObj(urlStr string) (tagging.Tagging, error) {
+	var err error
+
+	clnt, pErr := newClient(urlStr)
+	if pErr != nil {
+		console.Errorln(pErr.ToGoError().Error)
+		fatalIf(probe.NewError(errors.New("Unable to show tags")), "Unable to obtain client from provided url "+urlStr)
+	}
+	s3c, ok := clnt.(*s3Client)
+	if !ok {
+		fatalIf(probe.NewError(errors.New("Unable to show tags")), "Unable to show tags for object in URL "+urlStr)
+	}
+
+	bucketName, objectName := s3c.url2BucketAndObject()
+	if bucketName == "" && objectName == "" {
+		return tagging.Tagging{}, errors.New("Bucket name & Object name cannot be empty")
+	}
+	tagXML, err := s3c.api.GetObjectTagging(bucketName, objectName)
+	if err != nil {
+		return tagging.Tagging{}, err
+	}
+	var tagObj tagging.Tagging
+	if err = xml.Unmarshal([]byte(tagXML), &tagObj); err != nil {
+		console.Errorln(err.Error() + ", Unable to initialize Object Tags for display.")
+		return tagging.Tagging{}, err
+	}
+
+	return tagObj, err
+}
+
+// Color scheme for tag display
+func setTagShowColorScheme() {
+	console.SetColor(tagRowTheme, color.New(color.FgWhite))
+	console.SetColor(tagMainHeader, color.New(color.FgCyan))
+	console.SetColor(tagResultsSuccess, color.New(color.Bold, color.FgGreen))
+	console.SetColor(tagResultsFailure, color.New(color.Bold, color.FgRed))
+}
+
+func checkShowTagSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelp(ctx, "show")
+		os.Exit(globalErrorExitStatus)
+	}
+}
+
+func showTagInfoFieldMultiple(kvpairs []tagging.Tag) {
+	padLen := 0
+	for _, kv := range kvpairs {
+		if len(kv.Key) > padLen {
+			padLen = len(kv.Key)
+		}
+	}
+	padLen = getTagSpacePad(padLen)
+	for idx := 0; idx < len(kvpairs); idx++ {
+		displayField := fmt.Sprintf("    %-*s:    %s ", padLen, kvpairs[idx].Key, kvpairs[idx].Value)
+		console.Println(console.Colorize(tagRowTheme, displayField))
+	}
+}
+
+func getTagSpacePad(padLen int) int {
+	if padLen%4 == 0 {
```

Comment:
Can this be done with a switch command? The values are when modulo division is done, it is either `0,1,2or3` Using switch is better to read.